inc/notificationtargetentity.class.php

```diff
@@ -0,0 +1,161 @@
+          'expired' => __('Expiration date')
+      ];
+   }
+
+   public function addDataForTemplate($event, $options = []) {
+      global $CFG_GLPI, $DB;
+
+      $this->data['##credit.voucher##'] = $this->obj->getField('name');
+      $this->data['##credit.quantity##'] = $this->obj->getField('quantity');
+      $this->data['##credit.enddate##'] = $this->obj->getField('end_date');
+
+      $query='SELECT SUM(`glpi_plugin_credit_tickets`.`consumed`) AS `consumed_total`
+              FROM `glpi_plugin_credit_tickets` 
+              WHERE `glpi_plugin_credit_tickets`.`plugin_credit_entities_id` = '.$this->obj->getField('id');
+      $result = $DB->query($query);
+      $data = $DB->fetch_assoc($result);
+      $this->data['##credit.left##']=(int)$this->obj->getField('quantity')-(int)$data['consumed_total'];
+
+      $this->data['##lang.credit.enddate##']=__('End date');
+      $this->data['##lang.credit.left##']=__('Quantity remaining', 'credit');
+      $this->data['##lang.credit.quantity##']=__('Quantity sold', 'credit');
+      $this->data['##lang.credit.voucher##']=__('Credit voucher', 'credit');
+   }
+
+   public function getTags() {
+      $tags=[
+         'credit.voucher' => __('Credit voucher', 'credit'),
+         'credit.quantity' => __('Quantity sold', 'credit'),
+         'credit.enddate' => __('End date'),
+         'credit.left'    => __('Quantity remaining', 'credit'),
+      ];
+
+      foreach ($tags as $tag => $label) {
+         $this->addTagToList([
+            'tag'   => $tag,
+            'label' => $label,
+            'value' => true,
+         ]);
+      }
+
+      asort($this->tag_descriptions);
+   }
+
+
+   public static function install(Migration $migration) {
+      global $DB;
+
+      $migration->displayMessage("Migrate PluginCreditEntity notifications");
+
+      $template     = new NotificationTemplate();
+      $translation  = new NotificationTemplateTranslation();
+      $notification = new Notification();
+      $n_n_template = new Notification_NotificationTemplate();
+
+      $templates_id = false;
+      $query_id     = "SELECT `id`
+                       FROM `glpi_notificationtemplates`
+                       WHERE `itemtype`='PluginCreditEntity'
+                       AND `name` = 'Credit expired'";
+      $result       = $DB->query($query_id) or die ($DB->error());
+
+      if ($DB->numrows($result) > 0) {
+         $templates_id = $DB->result($result, 0, 'id');
+      } else {
+         $tmp = [
+            'name'     => 'Credit expired',
+            'itemtype' => 'PluginCreditEntity',
+            'date_mod' => $_SESSION['glpi_currenttime'],
+            'comment'  => '',
+            'css'      => '',
+         ];
+         $templates_id = $template->add($tmp);
+      }
+
+      if ($templates_id) {
+         if (!countElementsInTable($translation->getTable(), ['notificationtemplates_id' => $templates_id])) {
+
+            $contentText = 'The credit ##credit.voucher## will expire soon.';
```

Comment:
We used to put texts in specific tags instead of putting them into notification templates translations. Doing this makes default translation usable for any language (translation are made to be used when using custom labels or changing content between languages).

You can do it like here https://github.com/glpi-project/glpi/blob/9.3/bugfixes/inc/notificationtargetuser.class.php#L120.